File: Data Structures and Algorithms/49-GroupAnagrams/49-GroupAnagrams.py
# Last updated: 6/4/2026, 6:36:36 PM
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result = defaultdict(list)
        
        for s in strs:
            count = [0] * 26
            for c in s:
                count[ord(c) - ord('a')] += 1

            result[tuple(count)].append(s)

        return list(result.values())
        

File: Data Structures and Algorithms/49-GroupAnagrams/49-GroupAnagrams.cpp
// Last updated: 6/4/2026, 6:36:45 PM
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> ans;
        for(string& s: strs)
        {
            string key = s;
            sort(key.begin(),key.end());
            ans[key].emplace_back(s);
        }
        vector<vector<string>> result;
        for( auto entry: ans)
        {
            result.emplace_back(entry.second);
        }
        return result;
    }
};